lua/pl/pretty.lua

```diff
@@ -284,10 +284,11 @@ function pretty.write (tbl,space,not_clever)
                end
             end
             table.sort(ordered_keys, function (a, b)
-                if type(a) == type(b)  and type(a) == 'string' then
-                    return a < b
+                if type(a) == type(b) then
+                    return tostring(a) < tostring(b)
```

Comment:
What if the types here are not strings? Does this function handle tables with tables for keys properly? For example if they are both tables but they implement a `:__lt()` meta method casting them to a string to compare would be incorrect. Also casting numbers to strings to compare values is a waste of several cycles. Lets fix the return value logic bug without slowing this down with more casts than may be needed.

Reply:
Almost no issue when comparing non string keys.

Sorting key table is working fine, the `tostring()` is there to sort them using the ref (something like 'table: 0x558427c1b670').

As explained in #293, the goal of this sorting function is to obtain consistent output mostly for test purposes. Thus ignoring `:__lt()` meta method should not be an issue.

But there is one issue with table implementing the `:__tostring()` meta method, this call could be very slow.

Same issues with userdata type.

There is no key with number type in the `ordered_keys` table. They are excluded at line 282.

Reply:
Hi @alerque , sorry for pinging you again, but as the current behavior is making the lib crash, I’d like to provide a fix rather quickly.

Do you think I should do as I suggested just before ? I.e. searching if the object has a `__lt()` defined (and visible), and if so, use it before the piece of code I changed ?

Reply:
Given there are no number keys involved here it's probably fine how it is, but as the defactor maintainer I think @Tieske will have to be the one to make a final call on this and handle the merge if appropriate.